packages/did-auth-siop-op-authenticator/__tests__/shared/didAuthSiopOpAuthenticatorAgentLogic.ts

```diff
@@ -0,0 +1,243 @@
+    didDocumentMetadata: {},
+  },
+  issuer: did,
+  signer: {
+    id: did,
+    type: 'authentication',
+    controller: did,
+  },
+  jwt: 'ey...',
+  payload: {
+    scope,
+    response_type: ResponseType.ID_TOKEN,
+    client_id: did,
+    redirect_uri: redirectUrl,
+    response_mode: ResponseMode.POST,
+    response_context: ResponseContext.RP,
+    nonce: nonce,
+  },
+  verifyOpts: {
+    verification: {
+      mode: VerificationMode.INTERNAL,
+      resolveOpts: {},
+    },
+  },
+}
+
+export default (testContext: {
+  getAgent: () => ConfiguredAgent
+  setup: () => Promise<boolean>
+  tearDown: () => Promise<boolean>
+  isRestTest: boolean
+}) => {
+  describe('DID Auth SIOP OP Authenticator Agent Plugin', () => {
+    let agent: ConfiguredAgent
+
+    beforeAll(() => {
+      testContext.setup()
+      agent = testContext.getAgent()
+
+      nock(redirectUrl).get(`?stateId=${stateId}`).times(5).reply(200, requestResultMockedText)
+
+      const mockedParseAuthenticationRequestURIMethod = jest.fn()
+      OP.prototype.parseAuthenticationRequestURI = mockedParseAuthenticationRequestURIMethod
+      mockedParseAuthenticationRequestURIMethod.mockReturnValue(Promise.resolve(authenticationRequest))
+
+      const mockedVerifyAuthenticationRequestMethod = jest.fn()
+      OP.prototype.verifyAuthenticationRequest = mockedVerifyAuthenticationRequestMethod
+      mockedVerifyAuthenticationRequestMethod.mockReturnValue(Promise.resolve(authenticationVerificationMockedResult))
+
+      const mockedCreateAuthenticationResponse = jest.fn()
+      OP.prototype.createAuthenticationResponse = mockedCreateAuthenticationResponse
+      mockedCreateAuthenticationResponse.mockReturnValue(Promise.resolve(createAuthenticationResponseMockedResult))
+
+      const mockSubmitAuthenticationResponseMethod = jest.fn()
+      OP.prototype.submitAuthenticationResponse = mockSubmitAuthenticationResponseMethod
+      mockSubmitAuthenticationResponseMethod.mockReturnValue(Promise.resolve({ status: 200, statusText: 'example_value' }))
+
+      const mockedSelectVerifiableCredentialsForSubmissionMethod = jest.fn()
+      PresentationExchange.prototype.selectVerifiableCredentialsForSubmission = mockedSelectVerifiableCredentialsForSubmissionMethod
+      mockedSelectVerifiableCredentialsForSubmissionMethod.mockReturnValue(Promise.resolve({ errors: [], matches: ['match'] }))
+
+      const mockedSubmissionFromMethod = jest.fn()
+      PresentationExchange.prototype.submissionFrom = mockedSubmissionFromMethod
+      mockedSubmissionFromMethod.mockReturnValue(Promise.resolve({}))
+    })
+
+    afterAll(testContext.tearDown)
+
+    it('should authentication with DID SIOP without custom approval', async () => {
+      const result = await agent.authenticateWithDidSiop({
+        stateId,
+        redirectUrl,
+        didMethod,
+      })
+
+      expect(result.status).toEqual(200)
+    })
+
+    it('should authentication with DID SIOP with custom approval', async () => {
```

Comment:
-> authenticate

That is true for all it/test strings

Reply:
fixed